src/pools/fixedRate/FixedRateExchange.ts

```diff
@@ -362,13 +362,17 @@ export class FixedRateExchange {
     exchangeId: string,
     newRate: string
   ): Promise<TransactionReceipt> {
+    const exchange = await this.getExchange(exchangeId)
     const estGas = await this.estSetRate(
       address,
       exchangeId,
-      this.web3.utils.toWei(String(newRate))
+      await this.amountToUnits(exchange.baseToken, newRate.toString())
     )
     const trxReceipt = await this.contract.methods
-      .setRate(exchangeId, this.web3.utils.toWei(newRate))
+      .setRate(
+        exchangeId,
+        await this.amountToUnits(exchange.baseToken, newRate.toString())
```

Comment:
also should we have setNewRate in wei even if we have have the rate in raw string value in FreCreationParams used to create a new fixed rate for example ?

Reply:
nevermind we convert it there also here https://github.com/oceanprotocol/ocean.js/blob/v4main/src/utils/ContractUtils.ts#L83